Wrappers/Python/test/test_Operator.py

```diff
@@ -110,7 +111,48 @@ def test_ScaledOperator(self):
         scalar = 0.5
         sid = scalar * Identity(ig)
         numpy.testing.assert_array_equal(scalar * img.as_array(), sid.direct(img).as_array())
+    
+    def test_DiagonalOperator(self):
+        print ("test_DiagonalOperator")
+        
+        M = 3
+        ig = ImageGeometry(M, M)
+        x = ig.allocate('random_int')
+        diag = ig.allocate('random_int')
+        
+        # Set up example DiagonalOperator
+        D = DiagonalOperator(diag)
+    
+        # Apply direct and check whether result equals diag*x as expected.
+        z = D.direct(x)
+        numpy.testing.assert_array_equal(z.as_array(), (diag*x).as_array())
+    
+        # Apply adjoint and check whether results equals diag*(diag*x) as expected.
+        y = D.adjoint(z)
+        numpy.testing.assert_array_equal(y.as_array(), (diag*(diag*x)).as_array())
         
+    def test_MaskOperator(self):
+        print ("test_MaskOperator")
+        
+        M = 3
+        ig = ImageGeometry(M, M)
+        x = ig.allocate('random_int')
```

Comment:
In this case it makes sense that the mask is of `int` type, see #535.

Reply:
I actually wanted the mask to be of Boolean type rather than numerical 1.0 and 0.0, but didn't know whether that was possible.

Reply:
yes, you can use `dtype=numpy.bool` when allocating.